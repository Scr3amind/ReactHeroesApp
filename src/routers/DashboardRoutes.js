import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <div >
            <div className="w-screen h-screen bg-comic-pattern bg-repeat bg-contain opacity-30 absolute z-0"></div>
            <Navbar />
            <div className="relative z-10">
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroeId" component={HeroScreen} />
                    <Route exact path="/dc" component={DcScreen} />
                    <Route exact path="/search" component={SearchScreen} />

                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </div>
    )
}
