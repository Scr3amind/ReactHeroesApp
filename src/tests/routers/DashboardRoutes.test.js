import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import '../../setupTest';

describe('Pruebas en <DashboardRoutes />', () => {

    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: true,
            name: 'Isaac'
        }
    }
    
    test('debe mostrarse correctamente', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();    
    })
    
    
})
