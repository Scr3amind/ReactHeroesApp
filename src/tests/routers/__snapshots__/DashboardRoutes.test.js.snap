// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> debe mostrarse correctamente 1`] = `
<div>
  <div
    className="w-screen h-screen bg-comic-pattern bg-repeat bg-contain opacity-30 absolute z-0"
  />
  <nav
    className="bg-gradient-to-r from-gray-700 to-black shadow-2xl relative z-10 py-4 animate__animated animate__fadeInDown"
  >
    <div
      className="container mx-auto"
    >
      <div
        className="sm:flex justify-around"
      >
        <a
          className="text-white text-3xl font-bold p-3"
          href="/"
          onClick={[Function]}
        >
          Heroes App
        </a>
        <ul
          className="text-gray-400 sm:self-center text-xl border-t sm:border-none"
        >
          <li
            className="sm:inline"
          >
            <a
              aria-current="page"
              className="p-3 hover:text-white text-white font-bold"
              href="/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
          </li>
          <li
            className="sm:inline"
          >
            <a
              aria-current={null}
              className="p-3 hover:text-white"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
          </li>
          <li
            className="sm:inline"
          >
            <a
              aria-current={null}
              className="p-3 hover:text-white"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="sm:inline"
          >
            <button
              className="p-3 hover:text-white"
              onClick={[Function]}
            >
              Logout
            </button>
          </li>
          <li
            className="sm:inline"
          >
            <span
              className="px-3 text-gray-700 bg-white rounded-lg bg-opacity-80 font-bold"
            >
              Isaac
            </span>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="relative z-10"
  >
    <div
      className="flex flex-col items-center"
    >
      <h1
        className="text-2xl font-bold my-4"
      >
        Marvel
      </h1>
      <div
        className="flex"
      >
        <div
          className="flex gap-4 flex-wrap justify-center"
        >
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Spider Man"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Spider Man
              </p>
              <p>
                Peter Parker
              </p>
              <p
                className="text-xs"
              >
                Amazing Fantasy #15
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Captain America"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Captain America
              </p>
              <p>
                Steve Rogers
              </p>
              <p
                className="text-xs"
              >
                Captain America Comics #1
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Iron Man"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Iron Man
              </p>
              <p>
                Tony Stark
              </p>
              <p
                className="text-xs"
              >
                Tales of Suspense #39
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Thor"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Thor
              </p>
              <p>
                Thor Odinson
              </p>
              <p
                className="text-xs"
              >
                Journey into Myster #83
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Hulk"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Hulk
              </p>
              <p>
                Bruce Banner
              </p>
              <p
                className="text-xs"
              >
                The Incredible Hulk #1
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Wolverine"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Wolverine
              </p>
              <p>
                James Howlett
              </p>
              <p
                className="text-xs"
              >
                The Incredible Hulk #180
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Daredevil"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Daredevil
              </p>
              <p>
                Matthew Michael Murdock
              </p>
              <p
                className="text-xs"
              >
                Daredevil #1
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Hawkeye"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Hawkeye
              </p>
              <p>
                Clinton Francis Barton
              </p>
              <p
                className="text-xs"
              >
                Tales of Suspense #57
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Cyclops"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Cyclops
              </p>
              <p>
                Scott Summers
              </p>
              <p
                className="text-xs"
              >
                X-Men #1
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
          <div
            className="flex items-center justify-around h-52 w-72 shadow-lg rounded-xl bg-gradient-to-r from-gray-100 to-gray-200 animate__animated animate__fadeIn"
          >
            <div
              className="flex justify-center w-1/2"
            >
              <img
                alt="Silver Surfer"
                className="rounded-md shadow-lg h-48"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="flex flex-col justify-around h-full w-1/2 text-center"
            >
              <p
                className="font-bold"
              >
                Silver Surfer
              </p>
              <p>
                Norrin Radd
              </p>
              <p
                className="text-xs"
              >
                The Fantastic Four #48
              </p>
              <a
                className="text-blue-800"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
